parlai/core/torch_generator_agent.py

```diff
@@ -1611,7 +1718,17 @@ def select_paths(self, logprobs, prior_scores, current_length):
         # get the actual word id from residual of the same division
         tok_ids = best_idxs % voc_size
 
-        return (hyp_ids, tok_ids, best_scores)
+        tok_scores = torch.gather(logprobs, 1, tok_ids.unsqueeze(0)).view(-1)
+
+        tok_ranks = torch.arange(self.beam_size).to(logprobs.device) + 1
```

Comment:
Observe that `topk` returns elements in sorted order so the rank of an element in this returned list is simply its index plus 1.

Reply:
Ah right, can we just do `tok_ranks = logprobs.view(-1).argsort().argsort()`?